### PR TITLE
add service account resource

### LINKED ISSUE #125: [Feature Request] Add Service Account resource support
### Is your feature request related to a problem? Please describe.

Manage service accounts using terraform provider

### PR DESCRIPTION
**Pull request by kwadhwa-openai**
## What was changed

Adding Service Account CRUD support

## Why?
Service Accounts are currently managed using tcld. Adding support with terraform plugin will make it easy to maintain it in code

## Checklist

1. Closes #125 

2. How was this tested:
`make testacc`

3. Any docs updates needed?

generated the docs

### CODE REVIEW

**Comment by jlacefie** [comment]
TY for the contribution! 
Manually ran tests locally using make dye to #131 .  All tests passed. \o/
Only one nit, the generated docs don't have a sample in them.  Would you be able to add one, please?

```
RUN   TestAccBasicNamespace
=== PAUSE TestAccBasicNamespace
=== RUN   TestAccBasicNamespaceWithCertFilters
=== PAUSE TestAccBasicNamespaceWithCertFilters
=== RUN   TestAccNamespaceWithCodecServer
=== PAUSE TestAccNamespaceWithCodecServer
=== RUN   TestAccNamespaceRenameForcesReplacement
=== PAUSE TestAccNamespaceRenameForcesReplacement
=== RUN   TestAccNamespaceImport
=== PAUSE TestAccNamespaceImport
=== RUN   TestAccSpacesBetweenCertificateStrings
=== PAUSE TestAccSpacesBetweenCertificateStrings
=== RUN   TestAccNamespaceWithSearchAttributes
=== PAUSE TestAccNamespaceWithSearchAttributes
=== RUN   TestAccNamespaceImportSearchAttribute
=== PAUSE TestAccNamespaceImportSearchAttribute
=== RUN   TestAccNamespaceWithSearchAttributesUpdate
=== PAUSE TestAccNamespaceWithSearchAttributesUpdate
=== RUN   TestAccNamespaces
--- PASS: TestAccNamespaces (3.99s)
=== RUN   TestAccRegion
--- PASS: TestAccRegion (4.17s)
=== RUN   TestAccBasicServiceAccount
=== PAUSE TestAccBasicServiceAccount
=== RUN   TestAccBasicServiceAccountWithNamespaceAccesses
=== PAUSE TestAccBasicServiceAccountWithNamespaceAccesses
=== RUN   TestAccBasicUser
=== PAUSE TestAccBasicUser
=== RUN   TestAccBasicUserWithNamespaceAccesses
=== PAUSE TestAccBasicUserWithNamespaceAccesses
=== CONT  TestAccBasicNamespace
=== CONT  TestAccNamespaceImport
=== CONT  TestAccBasicNamespaceWithCertFilters
=== CONT  TestAccNamespaceWithCodecServer
=== CONT  TestAccBasicUserWithNamespaceAccesses
=== CONT  TestAccBasicUser
=== CONT  TestAccNamespaceImportSearchAttribute
=== CONT  TestAccNamespaceWithSearchAttributesUpdate
=== CONT  TestAccBasicServiceAccount
=== CONT  TestAccBasicServiceAccountWithNamespaceAccesses
--- PASS: TestAccBasicUser (12.21s)
=== CONT  TestAccNamespaceRenameForcesReplacement
--- PASS: TestAccBasicServiceAccount (12.25s)
=== CONT  TestAccNamespaceWithSearchAttributes
--- PASS: TestAccBasicNamespaceWithCertFilters (105.90s)
=== CONT  TestAccSpacesBetweenCertificateStrings
--- PASS: TestAccNamespaceWithSearchAttributesUpdate (112.80s)
--- PASS: TestAccNamespaceImport (121.79s)
--- PASS: TestAccNamespaceWithCodecServer (122.24s)
--- PASS: TestAccBasicServiceAccountWithNamespaceAccesses (122.86s)
--- PASS: TestAccBasicNamespace (123.71s)
--- PASS: TestAccBasicUserWithNamespaceAccesses (127.84s)
--- PASS: TestAccNamespaceImportSearchAttribute (134.69s)
--- PASS: TestAccNamespaceWithSearchAttributes (174.98s)
--- PASS: TestAccSpacesBetweenCertificateStrings (106.63s)
--- PASS: TestAccNamespaceRenameForcesReplacement (208.08s)
PASS
ok      github.com/temporalio/terraform-provider-temporalcloud/internal/provider        228.871s
=== RUN   TestCACertNormalization
--- PASS: TestCACertNormalization (0.00s)
=== RUN   TestUnknownEncodedCAValue
--- PASS: TestUnknownEncodedCAValue (0.00s)
=== RUN   TestNullEncodedCAValue
--- PASS: TestNullEncodedCAValue (0.00s)
=== RUN   TestEncodedCAValue
--- PASS: TestEncodedCAValue (0.00s)
PASS
ok      github.com/temporalio/terraform-provider-temporalcloud/internal/types   (cached)
```

**Comment by kwadhwa-openai** [comment]
@jlacefie I added the sample to this PR! I will also add it for other PRs

File: docs/resources/service_account.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+## Schema
+
+### Required
+
+- `account_access` (String) The role on the account. Must be one of [admin, developer, read] (case-insensitive)
+- `name` (String) The name associated with the service account.
+
+### Optional
+
+- `namespace_accesses` (Attributes List) The list of namespace accesses. (see [below for nested schema](#nestedatt--namespace_accesses))
```
Comment (swgillespie) [inline docs/resources/service_account.md:62]: not blocking feedback, but just so you know - we've had some issues (https://github.com/temporalio/terraform-provider-temporalcloud/issues/119, https://github.com/temporalio/terraform-provider-temporalcloud/issues/116, https://github.com/temporalio/terraform-provider-temporalcloud/issues/115) that have arisen from this tight coupling of users and namespace accesses as part of the same resource. I sketched out https://github.com/temporalio/terraform-provider-temporalcloud/pull/122 but I haven't finished the full idea yet, but my thinking here was to add another resource that represents a namespace access that can be manipulated even if the user object is referenced as a data source. Something we might also want to do here, too.

**Review by swgillespie — APPROVED** [review]
look great, thank you!

**Pull request merged**